packages/compass-aggregations/src/modules/saved-pipeline.ts

```diff
@@ -112,28 +112,33 @@ export const updatePipelineList =
 /**
  * Restore pipeline by an ID
  */
-export const openPipelineById = (
-  id: string
-): PipelineBuilderThunkAction<Promise<void>> => {
-  return async (dispatch, getState, { pipelineBuilder, pipelineStorage }) => {
+export const openStoredPipeline = (
```

Comment:
In every place we open stored pipeline we already loaded pipeline from disk, I changed this method to accept the whole stored pipeline meta so that we don't read data twice and can synchronously load pipeline removing some UI lag